File: src/Cadastro.API/Models/Request/FuncionarioRequest.cs
using System;
using System.Collections.Generic;

namespace Cadastro.API.Models.Request
{
    public class FuncionarioRequest
    {
        public FuncionarioRequest()
        {
            Telefones = new List<TelefoneRequest>();
        }
        public string UserId { get; set; }
        public string Matricula { get; set; }
        public string Cargo { get; set; }
        public string Nome { get; set; }
        public string SobreNome { get; set; }
        public string Email { get; set; }
        public DateTime DataNascimento { get; set; }
        public List<TelefoneRequest> Telefones { get; set; }
        public EnderecoRequest EnderecoComercial { get; set; }
        public EnderecoRequest EnderecoResidencial { get; set; }
        public bool Ativo { get; set; }

    }
}

File: src/cadastro.angular/src/app/models/request/funcionarioRequest.ts
import { FuncionarioResponse } from '../response/funcionarioResponse';
import { EnderecoRequest } from './enderecoRequest';
import { TelefoneRequest } from './telefoneRequest';

export class FuncionarioRequest {
    userId!: string;
    matricula!: string;
    cargo!: string;
    nome!: string;
    sobreNome!: string;
    email!: string;
    dataNascimento!: Date | null;
    telefones!: TelefoneRequest[];
    enderecoComercial!: EnderecoRequest;
    enderecoResidencial!: EnderecoRequest;
    ativo!: boolean;

   
    constructor (funcionario: FuncionarioResponse){
this.ativo = funcionario.ativo;
this.cargo = funcionario.cargo;
this.dataNascimento = funcionario.dataNascimento;
this.email = funcionario.email;
this.matricula = funcionario.matricula;
this.nome =  funcionario.nome;
this.sobreNome = funcionario.sobreNome;
this.userId = funcionario.userId;
this.enderecoComercial = new EnderecoRequest(funcionario.enderecoComercial);

    }

}
